webnn/resources/test_data/concat.json

```diff
@@ -2,662 +2,1712 @@
   "tests": [
     // concat 1D tensors
     {
-      "name": "concat two 1D tensors of same shape along axis 0",
-      "inputs": {
-        "shape": [[12], [12]],
-        "data": "float64Data"
-      },
+      "name": "concat two 1D of same shape along axis 0",
+      "inputs": [
+        {
+          "name": "input1",
+          "shape": [12],
+          "data": [
+            -0.39444134019222243,
+            0.8619825316530809,
+            0.3379962524218807,
+            -0.9906398615400507,
+            0.576785657225761,
+            0.3227640108329237,
+            -0.44735022799701873,
+            0.11028251232581932,
+            -0.5945112749179908,
+            -0.40284849555754754,
+            -0.9531654171044694,
+            -0.6731740531810844
+          ]
+        },
+        {
+          "name": "input2",
+          "shape": [12],
+          "data": [
+            0.4918989118791477,
+            -0.15864110312378976,
+            -0.34188115459083157,
+            -0.9158143500894873,
+            -0.7206121708970712,
+            -0.7993468785008635,
+            0.6653799854931952,
+            0.03886038855553897,
+            0.5182055416768865,
+            -0.8742016938344297,
+            -0.479021891130635,
+            0.1211843166661235
+          ]
+        }
+      ],
       "axis": 0,
       "expected": {
         "shape": [24],
-        "data": "float32Axis0"
+        "data": [
+          -0.3944413363933563,
+          0.861982524394989,
+          0.337996244430542,
+          -0.990639865398407,
+          0.576785683631897,
+          0.32276400923728943,
+          -0.44735023379325867,
+          0.11028251051902771,
+          -0.5945112705230713,
+          -0.402848482131958,
+          -0.9531654119491577,
+          -0.6731740236282349,
+          0.49189892411231995,
+          -0.15864109992980957,
+          -0.3418811559677124,
+          -0.9158143401145935,
+          -0.7206121683120728,
+          -0.7993468642234802,
+          0.6653800010681152,
+          0.03886038810014725,
+          0.5182055234909058,
+          -0.8742017149925232,
+          -0.4790218770503998,
+          0.1211843192577362
+        ]
       },
       "type": "float32"
```

Comment:
You may need to set type for each operand. Because some operations may have different types for different input and output operands.

Reply:
Thanks, I will update it. And is it ok to still keep "type" here to indicate that current test is for float32 precision?

Reply:
What's the use case of it? The spec only defines operand data type, not a operation data type.

Reply:
Regarding to your last comment "some operations may have different types for different input and output operands.", which is the main precision target. 
This helper "type" would clearly show the test precision target.

Reply:
I understand the precision target would be represented by the data type of the expected output data. That is accurate and will be used in the test.

Unless this helper "type" has a specific use case, I'd recommend to remove it.